vignettes/extract-bias.Rmd

```diff
@@ -0,0 +1,494 @@
+  n_samples <- parameters_range[params_idx, "n_samples"]
+
+  if (dist == "lnorm") {
+    true_median <- do.call(
+      paste0("q", dist),
+      list(
+        p = 0.5,
+        meanlog = parameters_range[params_idx, "param_1"],
+        sdlog = parameters_range[params_idx, "param_2"]
+      )
+    )
+    true_range <- do.call(
+      paste0("r", dist),
+      list(
+        n = n_samples,
+        meanlog = parameters_range[params_idx, "param_1"],
+        sdlog = parameters_range[params_idx, "param_2"]
+      )
+    )
+    true_range <- c(min(true_range), max(true_range))
+  } else {
+    true_median <- do.call(
+      paste0("q", dist),
+      list(
+        p = 0.5,
+        shape = parameters_range[params_idx, "param_1"],
+        scale = parameters_range[params_idx, "param_2"]
+      )
+    )
+    true_range <- do.call(
+      paste0("r", dist),
+      list(
+        n = n_samples,
+        shape = parameters_range[params_idx, "param_1"],
+        scale = parameters_range[params_idx, "param_2"]
+      )
+    )
+    true_range <- c(min(true_range), max(true_range))
+  }
+
+  true_values <- c(true_median, true_range)
+
+  # message about stochastic optimisation suppressed
+  estim <- suppressMessages(
+    replicate(
+      n = 5,
+      expr =  extract_param(
+        type = "range",
+        values = true_values,
+        distribution = dist,
+        samples = n_samples
+      )
+    )
+  )
+  estim_param_var[[params_idx]] <- apply(estim, MARGIN = 1, FUN = var)
+}
+
+# combine results
+results <- cbind(parameters_range, do.call(rbind, estim_param_var))
+
+colnames(results) <- c(
+  "dist", "param_1", "param_2", "n_samples", "estim_param_1_var",
+  "estim_param_2_var"
+)
+```
+
+```{r, plot-estim-var-range, class.source = 'fold-hide', fig.cap="Parameter extraction precision, facetted by distribution. Parameter 1 is either the shape parameter, for gamma and Weibull distributions, or meanlog for the lognormal distribution. Parameter 2 is either the scale parameter for gamma and Weibull distributions, or sdlog for the lognormal distribution."}
+ggplot(data = results) +
+  geom_point(mapping = aes(
+    x = estim_param_1_var,
+    y = estim_param_2_var,
+    colour = n_samples
+  )) +
+  scale_x_continuous(name = "Parameter 1 Variance") +
+  scale_y_continuous(name = "Parameter 2 Variance") +
+  labs(colour = "No. Samples") +
+  theme_bw() +
+  scale_color_viridis_c() +
+  facet_wrap(facets = vars(dist), scales = "free") +
+  theme(
+    strip.background = element_blank(),
+    axis.text.x = element_text(angle = 30, vjust = 0.5)
+  )
+```
+
+::: {.alert .alert-primary}
+From the plots in this vignette, the bias is low and precision is high
+when extracting parameters from the gamma, lognormal and Weibull distributions 
+from both percentiles of the distribution and from median and range of a data
+set.
+
+The asymmetry of percentiles and sample size of the data does not noticeably 
+influence the bias in parameter extraction.
```

Comment:
Figure 2 suggests bias for extraction from median + range can be substantial for smaller sample sizes? Which makes sense, given range has high variance as a summary statistic?

Reply:
Can you specify which part of the plot you mean? It seems for gamma and weibull that the sample size may have a small influence on bias, but for lnorm it is the largest sample sizes that have the largest biases in the sdlog parameter.